src/error-handling.cpp

```diff
@@ -49,28 +49,35 @@ namespace librealsense
          {
              try
              {
-
                  auto val = static_cast<uint8_t>(_option->query());
 
                  if (val != 0 && !_silenced)
                  {
                      auto strong = _notifications_processor.lock();
-                     if (strong) strong->raise_notification(_decoder->decode(val));
 
-                     val = static_cast<uint8_t>(_option->query());
-                     if (val != 0)
+                     // First reset the value in the FW.
+                     // Reading from last-error control is supposed to set it to zero in the
+                     // firmware If this is not happening there is some issue 
+                     // Note: if an error will be raised between the 2 queries, this will cause the error polling loop to stop
+                     auto reseted_val = static_cast<uint8_t>( _option->query() );
+                     if( reseted_val != 0 )
```

Comment:
This should still happen only AFTER we've notified of the original `val`, no?

Reply:
👍